Extensions/DialogueTree/dialoguetools.js

```diff
@@ -0,0 +1,538 @@
+  if (optionIndex >= this.options.length) optionIndex = this.options.length - 1;
+  if (optionIndex < 0) optionIndex = 0;
+  return optionIndex;
+};
+
+// Internal method to allow for cycling option selection
+gdjs.dialogueTree._cycledOptionIndex = function(optionIndex) {
+  if (optionIndex >= this.options.length) optionIndex = 0;
+  if (optionIndex < 0) optionIndex = this.options.length - 1;
+  return optionIndex;
+};
+
+/**
+ * Expression to get the text of an option the player can select.
+ * Use this with lineOptionsCount to render options for the player when a line of the Options type is parsed
+ * @param {number} optionIndex The index of the option you want to get
+ */
+gdjs.dialogueTree.getLineOption = function(optionIndex) {
+  if (!this.options.length) return [];
+  optionIndex = gdjs.dialogueTree._normalizedOptionIndex(optionIndex);
+  return this.options[optionIndex];
+};
+
+// Expression to get the number of options that are presented to the player during the parsing of an Options type line
+gdjs.dialogueTree.lineOptionsCount = function() {
+  if (this.options.length) {
+    return this.optionsCount;
+  }
+};
+
+// Action to confirm the currently selected by the plyer option during Options type line parsing.
+// This will advance the dialogue tree to whichever dialogue branch was selected by the player
+gdjs.dialogueTree.confirmSelectOption = function() {
+  if (
+    this.dialogueData.select &&
+    !this.selectedOptionUpdated &&
+    this.selectOption !== -1
+  ) {
+    this.commandCalls = [];
+    this.dialogueData.select(this.selectOption);
+    this.dialogueData = this.dialogue.next().value;
+    gdjs.dialogueTree.advanceDialogue();
+  }
+};
+
+// Action to Select next option during Options type line parsing. Hook this to your game input
+gdjs.dialogueTree.selectNextOption = function() {
+  if (this.dialogueData.select) {
+    this.selectOption += 1;
+    this.selectOption = gdjs.dialogueTree._cycledOptionIndex(this.selectOption);
+    this.selectedOptionUpdated = true;
+  }
+};
+
+// Action to Select previous option during Options type line parsing. Hook this to your game input
+gdjs.dialogueTree.selectPreviousOption = function() {
+  if (this.dialogueData.select) {
+    this.selectOption -= 1;
+    this.selectOption = gdjs.dialogueTree._cycledOptionIndex(this.selectOption);
+    this.selectedOptionUpdated = true;
+  }
+};
+
+/**
+ * Action to Select option by index during Options type line parsing.
+ * @param {number} optionIndex The index of the option you want to select
+ */
+gdjs.dialogueTree.selectOption = function(optionIndex) {
+  if (this.dialogueData.select) {
+    this.selectOption = gdjs.dialogueTree._normalizedOptionIndex(
+      this.selectOption
+    );
+    this.selectedOptionUpdated = true;
+  }
+};
+
+// Expression to get the currently selected option
+gdjs.dialogueTree.getSelectOption = function() {
+  if (this.dialogueData.select) {
+    return this.selectOption;
+  }
+};
```

Comment:
Same, be very careful about not returning something. If we had typescript or flow typing, you would certainly get an error - we don't have these in the game engine so we have to be very cautious about the safety of our methods.